packages/react/src/components/DatePickerInput/DatePickerInput.tsx

```diff
@@ -173,11 +174,22 @@ const DatePickerInput = React.forwardRef(function DatePickerInput(
     type = 'text',
     warn,
     warnText,
+    readOnly,
     ...rest
   } = props;
   const prefix = usePrefix();
   const { isFluid } = useContext(FormContext);
   const datePickerInputInstanceId = useId();
+
+  const normalizedProps = useNormalizedInputProps({
+    id,
+    readOnly,
+    disabled,
+    invalid: invalid === true,
+    invalidText,
+    warn: warn === true,
```

Comment:
```suggestion
    invalid,
    invalidText,
    warn,
```
This should be able to work without the conditions

Reply:
yes  I have made the changes can you check